libdnf/hy-query.c

```diff
@@ -1868,3 +1869,66 @@ hy_filter_duplicated(HyQuery query)
         add_duplicates_to_map(pool, query->result, samename, start_block, i);
     }
 }
+
+int
+hy_filter_unneeded(HyQuery query, DnfSwdb *swdb, const int debug_solver)
```

Comment:
Why debug_solver is int and not a bool? You want use more levels of debugging in the future or only on/off?